doc/userguide/devguide/internals/engines/stream/inspection_raw_data.rst

```diff
@@ -0,0 +1,117 @@
+Inspection of raw stream data
+#############################
+
+Stream Engine
+^^^^^^^^^^^^^
+
+Suricata's Stream Engine tracks and processes all the TCP stream data. Its responsibilities include
+
+* TCP segment reassembly
+* TCP data normalization
+* gap management and handling
+* maintaining internal caches
+* handling of special cases like TCP URG ptr
+* applying user-defined constraints like stream depth etc
+
+for IDS as well as inline mode.
+
+Internal storage of stream data
+===============================
+
+For a stream with small gaps, a Red Black Tree is used to store the streaming buffer blocks.
+
+For a stream with large gaps, regions (list of blocks of data) are used.
```

Comment:
What is small/large ?

Reply:
262144 bytes. Should I mention such details as well?

Reply:
maybe better to be precise, you can also point to STREAMING_BUFFER_REGION_GAP_DEFAULT